### DISCUSSION TITLE
Dexie 3.2 is out

### DISCUSSION THREAD

**Discussion by dfahlander** [dexie/Dexie.js#1436] [Show and tell]
# 3.2.0 released

At last, Dexie with [liveQuery()](https://dexie.org/docs/liveQuery()) is released as stable version which is a big step as samples and tutorials on the home page could be updated to recommend using the new reactive feature of Dexie.

With all the news in 3.2 it could have been argued it should be a new major version, but that could also make people more hesitated from upgrading, and since there are no breaking changes from 3.0.3 I decided to let it stay on the same major.

See [release notes](https://github.com/dfahlander/Dexie.js/releases/tag/v3.2.0)

**Comment by dfahlander** [comment]
All further releases on 3.x will originate from [master-3](https://github.com/dexie/Dexie.js/commits/master-3) as the main master branch is now reflecting the 4.0-track.

**Comment by jimmywarting** [comment]
keep coming back to dexie.js cuz it's one of the best idb out there!

I'm amazed to come back and see that it got support for reactive stuff, super cool!
just to be sure, this dose not depend on some native IndexedBD change event or something that i have missed?

An non dexie.js pice of code (that use native idb directly) wouldn't be able to trigger the dexie.js changes, right?

**Reply by dfahlander** [reply]
Thanks Jimmy and much welcome back! The reactiveness is Dexie-specific and requires Dexie on both the mutating and observing side as there's no reactive standard in IDB. There's a old proposal from Google [indexeddb observer](https://chromestatus.com/feature/5669292892749824) that was the closest proposal so far but seems not to have reached consensus.

**Reply by jimmywarting** [reply]
yea, i figured as such. i remember reading about some observer/change event proposal somewhere before.

**Comment by jimmywarting** [comment]
Also notice the new toClass stuff.
do you know what would be even more impressive?

making the class reactive with getter/setters as well using the Proxy and just be able to write stuff like:

```js
export class Friend extends Entity {
  id
  name
  age

  birthday() {
    this.age++
  }
}
```

**Reply by dfahlander** [reply]
Yes, unless it would be percieved as too magical as it could cause unwanted/ unexpected mutations. I haven't figured out yet whether it's good or not. Opinions welcome on this topic!
